lib/rust/lexer/definition/src/token.rs

```diff
@@ -216,6 +338,128 @@ impl Default for LineEnding {
 
 
 
+// =================
+// === TextStyle ===
+// =================
+
+/// The style of the text literal.
+#[derive(Copy,Clone,Debug,Eq,PartialEq)]
+pub enum TextStyle {
+    // === Line ===
+
+    /// A interpolated text line literal.
+    FormatLine,
+    /// A raw text line literal.
+    RawLine,
+    /// An unclosed text line literal.
+    UnclosedLine,
+
+    // === Inline Block ===
+
+    /// A format inline block text literal.
+    FormatInlineBlock,
+    /// A raw inline block text literal.
+    RawInlineBlock,
+
+    // === Block ===
+
+    /// An interpolated text block literal.
+    FormatBlock,
+    /// A raw text block literal.
+    RawBlock,
+}
+
+impl TextStyle {
+    /// Calculate the length of the delimiters for a particular style of text literal.
+    pub fn length(self) -> usize {
+        match self {
+            TextStyle::FormatLine        => lexeme::format_quote().length() * 2,
+            TextStyle::RawLine           => lexeme::raw_quote().length() * 2,
```

Comment:
If I understand correctly, this takes into consideration the starting and the ending quote, right?

Reply:
Yes.